src/shared/bpf-program-v2.c

```diff
@@ -0,0 +1,67 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include <bpf/bpf.h>
+#include <fcntl.h>
+
+#include "alloc-util.h"
+#include "bpf-program-v2.h"
+#include "fd-util.h"
+#include "memory-util.h"
+
+int bpf_program_v2_new(int fd, enum bpf_attach_type attach_type, BPFProgramV2 **ret) {
+        _cleanup_(bpf_program_v2_freep) BPFProgramV2 *p = NULL;
+
+        assert(ret);
+
+        p = new(BPFProgramV2, 1);
+        if (!p)
+                return log_oom();
```

Comment:
We don't normally log in "library-like" code - eg, stuff in src/shared. Return -ENOMEM here, and let the caller log.